tests/e2e-tests/specs/backend/all-products.test.js

```diff
@@ -3,20 +3,53 @@
  */
 import {
 	insertBlock,
-	getEditedPostContent,
-	createNewPost,
+	getAllBlocks,
 	switchUserToAdmin,
 } from '@wordpress/e2e-test-utils';
 
-describe( 'All Products', () => {
-	beforeEach( async () => {
+import { visitBlockPage } from '@woocommerce/blocks-test-utils';
+
+/**
+ * Internal dependencies
+ */
+
+const block = {
+	name: 'All Products',
+	slug: 'woocommerce/all-products',
+	class: '.wc-block-all-products',
+};
+
+describe( `${ block.name } Block`, () => {
+	beforeAll( async () => {
 		await switchUserToAdmin();
-		await createNewPost();
+		await visitBlockPage( `${ block.name } Block` );
+	} );
+
+	it( 'can only be inserted once', async () => {
+		await insertBlock( block.name );
+		expect( await getAllBlocks() ).toHaveLength( 1 );
 	} );
 
-	it( 'can be created', async () => {
-		await insertBlock( 'All Products' );
+	it( 'renders without crashing', async () => {
+		// Gutenberg error
+		expect(
+			( await page.content() ).match(
+				/Your site doesn’t include support for/gi
+			)
+		).toBeNull();
+		// Our ErrorBoundary
+		expect(
+			( await page.content() ).match(
+				/There was an error whilst rendering/gi
+			)
+		).toBeNull();
+		// Validation Error
+		expect(
+			( await page.content() ).match(
+				/This block contains unexpected or invalid content/gi
+			)
+		).toBeNull();
```

Comment:
looks like this is something that could be extracted to a helper at some point (not necessarily in this pull)